molecule_vagrant/modules/vagrant.py

```diff
@@ -508,79 +351,57 @@ def _get_config(self):
         return molecule.config.Config(molecule_file)
 
     def _write_vagrantfile(self):
+        # Helper lambdas to get the CPU and memory fields for each provider
+        cpu_field = (
+            lambda provider: "vmx['numvcpus']"
+            if provider.startswith("vmware_")
+            else "cpus"
+        )
+        memory_field = (
+            lambda provider: "vmx['memsize']"
+            if provider.startswith("vmware_")
+            else "memory"
+        )
+
```

Comment:
too bad that vmware needs a special case. I hope that vmware is the only one. At least with your change, instead of having to modify the ``Vagrantfile`` template for every provider, we only have to deal with exceptions, so that's nice.

Reply:
Full disclosure, I haven't tested this with VMWare. This logic was in the original code so I simply carried it over here.